File: intersection_of_two_arrays.rb
# @param {Integer[]} nums1
# @param {Integer[]} nums2
# @return {Integer[]}
def intersection(nums1, nums2)
  nums1.uniq! 
  nums2.uniq! 
  ans = []
  nums1.each do |x|
    nums2.reverse_each do |y|
      ans << nums2.delete(y) if  x == y
    end
  end
  ans
end

File: intersection-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
  var ans = [];
  var hash = {};
  for (var n1=0; n1<nums1.length; n1++) {
    if (!hash[nums1[n1]]) hash[nums1[n1]] = true;
  }
  
  for (var n2=0; n2<nums2.length; n2++) {
    if (hash[nums2[n2]]) ans.push(nums2[n2]);
  }
  
  return [...new Set(ans)];
};
